addons/mail/static/src/new/public/discuss_public.js

```diff
@@ -46,12 +43,23 @@ export class DiscussPublic extends Component {
                             }
                         });
                     }
+                } else if (isChannelTokenSecret) {
+                    this.removeChannelTokenFromURL();
```

Comment:
Do we still need this method to be present in useEffect, twice?
`useEffect` was used to detect change from welcome to channel, but in this case I feel like the change of URL could be done directly in the boot of the page, as soon as we have the necessary data to make the condition?

Reply:
You're right, updated :+1: